blog/2022-06-21-using-temporal-as-a-node-task-queue.md

```diff
@@ -0,0 +1,354 @@
+---
+tags:
+  - temporal
+  - community
+posted_on_: 2022-06-21T00:00:00Z
+slug: using-temporal-as-a-node-task-queue
+title: 'Using Temporal as a Node.js Task Queue'
+author: Valeri Karpov
+author_title: Community Member
+author_image_url: https://avatars.githubusercontent.com/u/1620265?v=4
+release_version: V1.15
+---
+
+<!--truncate-->
+
+There are numerous task queues for Node.js: [Bull](https://optimalbits.github.io/bull/) is the most popular, but [Bee Queue](https://www.npmjs.com/package/bee-queue), [Agenda](https://www.npmjs.com/package/agenda), and [Kue](https://www.npmjs.com/package/kue) are also common.
+All these task queue libraries solve a similar problem.
+They let you put a task onto a queue, and define a worker that consumes tasks from the queue.
+While Temporal is much more than a task queue, Temporal does solve the same problems as these task queue libraries.
+Plus, Temporal's feature set provides several advantages if you use Temporal as your task queue.
+
+## Setting Up a Task Queue With Temporal
+
+A common use case for a task queue is sending a welcome email when a new user registers.
+Sending email is a good candidate for a task queue, because sending an email often requires a slow API call and is more of a "side effect" of registering the user.
+The following is a basic example of defining an email queue with Bull.
```

Comment:
Add headings before each code snippet that indicate which library is being used

Reply:
I'm not sure this change is necessary, because there's only 2 code samples that use Bull. The rest are Temporal.